connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/errors/ConnectExceptionMapper.java

```diff
@@ -52,6 +52,7 @@ public Response toResponse(ConnectException exception) {
                     .build();
         }
 
+        log.error("Uncaught exception in REST call: ", exception);
```

Comment:
One more quick question: will we have enough context in the log to tie this to the specific HTTP request?

Reply:
You probably can't tie it to a specific request (other than proximity in the log to the Apache-format request log). You'll be able to tie it to a type of request just based on the stacktrace since it'll be in one of the handlers.

In general I'm not sure there's a good way to tie back to specific requests as the most unique thing we'd normally log is the URL path. I also am pretty sure normally the way this ends up getting used is to aid in debugging from the user side, so they usually know what's failing, just not why.

Reply:
Yeah, I was wondering if it would make sense to pass the request context so that at least you could add the URL path (and maybe params) to the message.

One additional nit by the way: when DEBUG is enabled, we'll see this message twice because of the log line above. We could print this only if debug level is not enabled or something.

Reply:
Ok, updated to log the path. I also realized that we don't need a separate handler to fix the non-JSON response when the type isn't a ConnectException, we can just change this ExceptionMapper to handle all exceptions, so I've included that fix as well.